clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -358,12 +358,19 @@ private void process(final ShareAcknowledgeAsyncEvent event) {
      * consumer join the share group if it is not part of it yet, or send the updated subscription if
      * it is already a member.
      */
-    private void process(final ShareSubscriptionChangeEvent ignored) {
+    private void process(final ShareSubscriptionChangeEvent event) {
         if (!requestManagers.shareHeartbeatRequestManager.isPresent()) {
```

Comment:
Just a side comment, but this state should be impossible? Can we fail the event with an exception? No one sees warnings.

For a future patch, I think we need to separate the processor for the share consumer.